tls/s2n_server_hello.c

```diff
@@ -152,14 +180,23 @@ int s2n_server_hello_send(struct s2n_connection *conn)
     /* Create the server random data */
     GUARD(s2n_stuffer_init(&server_random, &b));
 
-    r.data = s2n_stuffer_raw_write(&server_random, S2N_TLS_RANDOM_DATA_LEN);
-    r.size = S2N_TLS_RANDOM_DATA_LEN;
-    notnull_check(r.data);
-    GUARD(s2n_get_public_random_data(&r));
+    rand.data = s2n_stuffer_raw_write(&server_random, S2N_TLS_RANDOM_DATA_LEN);
+    rand.size = S2N_TLS_RANDOM_DATA_LEN;
+    notnull_check(rand.data);
+    GUARD(s2n_get_public_random_data(&rand));
 
     protocol_version[0] = (uint8_t)(conn->actual_protocol_version / 10);
     protocol_version[1] = (uint8_t)(conn->actual_protocol_version % 10);
 
+    /* Protect against downgrade attacks according to RFC 8446 section 4.1.3 */
```

Comment:
Might be worth moving all the random data work here to its own function to keep send clean(er)

Reply:
Refactored in the next push.